opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -155,25 +155,55 @@ message MetricDescriptor {
   repeated string label_keys = 5;
 }
 
+// LabelValues is a list of label string values.
+message LabelValues {
+  // Values of labels. Each element in this array is the value of the key defined at the
+  // corresponding index of MetricDescriptor.label_keys array. The number of elements
+  // in these 2 arrays must be the same. Note that values array must include empty
+  // strings and must also contain an empty string value for labels for which the
+  // value was originally unspecified.
+  repeated string values = 1;
+
+  // Some metric sources allow setting the value of a label to be equal not just to
+  // empty string but to an unspecified value, or in other words the value of a particular
+  // label defined in label_keys array may be missing in a particular timeseries.
+  // In order to support this concept and because ProtoBuf cannot distinguish empty string
+  // from unspecified string we store indexes of unspecified labels in "values" array in
+  // this field. This field contains only indexes of unspecified labels.
+  // This field should not contain duplicate numbers, numbers less than zero or
+  // numbers greater than or equal to the length of values array.
+  repeated int32 unspecified_value_indexes = 2;
```

Comment:
should the array be ordered to simplify merging?

Reply:
I am trying to think of a usecase when such merging is needed for the proto data and fail to find one. This data typically either gets generated in client libraries based on in-memory representation of the metric as-is or is received by backends and stored in a way that is likely decoupled from this proto representation completely. So, basically this array is a likely a write-once, read-once, immutable data.

Even if there is an obscure use case this array is likely to be mostly empty or contain at most a few elements. Merging them is going to be fast performance-wise even if it is unsorted.